### ISSUE TITLE
Fix evaluate function for single-class mode

### ISSUE THREAD

**Issue by valhassan** [NRCan/geo-deep-learning#394]
We would like to apply a sigmoid function on output logits and threshold (e.g. output_arr > threshold) before calculating metrics with corresponding labels. This assumes the model does not use a sigmoid layer before outputs are returned.

For reference this will affect lines https://github.com/NRCan/geo-deep-learning/blob/8678cd900f12d2ded72395131b804c3babd74474/train_segmentation.py#L455-L463
